api/users/views.py

```diff
@@ -662,3 +662,20 @@ def perform_destroy(self, instance):
             if val['id'] in current_institutions:
                 user.remove_institution(val['id'])
         user.save()
+
+class UserPublicFiles(JSONAPIBaseView, generics.ListAPIView, UserMixin, ODMFilterMixin):
+    view_name = 'user-public-files'
+    view_category = 'users'
+    serializer_class = PublicFilesSerializer
+
+    required_read_scopes = [CoreScopes.USERS_READ, CoreScopes.NODE_REGISTRATIONS_READ]
+    required_write_scopes = [CoreScopes.USERS_WRITE, CoreScopes.NODE_REGISTRATIONS_WRITE]
+
+    permission_classes = (
+        drf_permissions.IsAuthenticatedOrReadOnly,
```

Comment:
You may want to define a custom permissions class that ensures authenticated users who aren't the requested user cannot write.

Also, it seems somewhat counter-intuitive that a *public* endpoint requires authentication.